pkg/controller/mysqlcluster/mysqlcluster_controller.go

```diff
@@ -171,22 +172,27 @@ func (r *ReconcileMysqlCluster) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err
 	}
 
-	configMapResourceVersion := configMapSyncer.GetObject().(*corev1.ConfigMap).ResourceVersion
-	secretResourceVersion := secretSyncer.GetObject().(*corev1.Secret).ResourceVersion
+	cmRV := configMapSyncer.GetObject().(*corev1.ConfigMap).ResourceVersion
+	sRV := secretSyncer.GetObject().(*corev1.Secret).ResourceVersion
 
 	// run the syncers for services, pdb and statefulset
 	syncers := []syncer.Interface{
 		mysqlcluster.NewHeadlessSVCSyncer(cluster),
 		mysqlcluster.NewMasterSVCSyncer(cluster),
 		mysqlcluster.NewHealthySVCSyncer(cluster),
 
-		mysqlcluster.NewStatefulSetSyncer(cluster, configMapResourceVersion, secretResourceVersion, r.opt),
+		mysqlcluster.NewStatefulSetSyncer(cluster, cmRV, sRV, r.opt),
 	}
 
 	if len(cluster.Spec.MinAvailable) != 0 {
 		syncers = append(syncers, mysqlcluster.NewPDBSyncer(cluster))
 	}
 
+	// add pods syncers for every node status
+	for _, ns := range cluster.Status.Nodes {
+		syncers = append(syncers, mysqlcluster.NewPodSyncer(cluster, ns.Name))
```

Comment:
We need to first set the `role=slave` label and then set the `role=master` to the actual master so that the master is switched atomically. We can do this by making the master node syncer to be the last syncer in list.